### PR TITLE
Tech task: [#162322] Fix flaky specs

### PR DESCRIPTION
**Pull request by LeticiaErrandonea**
# Release Notes
* Tech task: Fix flaky specs

### CODE REVIEW

File: spec/system/admin/facility_orders_search_spec.rb [hunk 1]
```diff
@@ -99,12 +99,12 @@
           expect(page).to have_css(".fa-users", count: 1) # cross_core_orders.last
           expect(page).to have_css(".fa-building", count: 1) # cross_core_order_originating_facility
 
-          expect(page).to have_content(cross_core_order_originating_facility.id)
-          expect(page).to have_content(cross_core_orders.last.id)
+          expect(page).to have_link(cross_core_order_originating_facility.id, href: facility_order_path(cross_core_order_originating_facility.facility, cross_core_order_originating_facility))
```
Comment (giladshanan) [inline spec/system/admin/facility_orders_search_spec.rb:102]: Were you able to figure out why this was failing?  Why does specifying the href fix it?

Maybe we can set these expected hrefs in variable to make this more readable?
Also, `cross_core_order_originating_facility` is an order, so how about we call it `originating_cross_core_order`?

File: spec/system/admin/adding_to_a_cross_core_order_spec.rb [hunk 2]
```diff
@@ -219,6 +219,8 @@
         expect(page).not_to have_content("Cross Core Project ID")
         expect(page.has_selector?("option", text: cross_core_product_facility.name, visible: false)).to be(true)
         expect(page.has_selector?("option", text: product.name, visible: false)).to be(true)
+
+        # I think we can comment this spec, as it isn't clear why it fails from time to time.
```
Comment (giladshanan) [inline spec/system/admin/adding_to_a_cross_core_order_spec.rb:223]:
```suggestion
        # TODO: I think we can comment this spec, as it isn't clear why it fails from time to time.
```

File: spec/system/admin/instrument_schedule_positions_spec.rb [hunk 3]
```diff
@@ -25,6 +25,8 @@
   describe "as a director" do
     let(:user) { create(:user, :facility_director, facility: facility) }
 
+    # Some of these specs have been failing even though the arrays look the same in the error message.
```
Comment (giladshanan) [inline spec/system/admin/instrument_schedule_positions_spec.rb:28]: Should this be a `TODO:` comment?
Where is the `appear_in_order` matcher defined?

File: spec/system/admin/all_transactions_search_spec.rb [hunk 4]
```diff
@@ -27,14 +27,15 @@
...
     select_from_chosen accounts.second.account_list_item, from: "Payment Sources"
     click_button "Filter"
-    
+
     expect(page).to have_content("Total")
-    expect(page).not_to have_link(order_detail.id.to_s)
-    expect(page).to have_link(orders.second.order_details.first.id.to_s)
+    expect(page).not_to have_link(order_detail.id.to_s, href: manage_facility_order_order_detail_path(order_detail.facility, order_detail.order, order_detail))
+    expect(page).to have_link(orders.second.order_details.first.id.to_s, href: manage_facility_order_order_detail_path(orders.second.facility, orders.second, orders.second.order_details.first))
+    # manage-order-detail
```
Comment (giladshanan) [inline spec/system/admin/all_transactions_search_spec.rb:34]: Can you add some more detail to this comment?

**Review by giladshanan — COMMENTED** [review]
Looks like good progress, I left some comments/questions

File: spec/system/admin/adding_to_a_cross_core_order_spec.rb (hunk 2 above)
Comment (LeticiaErrandonea) [reply spec/system/admin/adding_to_a_cross_core_order_spec.rb:223]: Oh, I wasn't clear here. I meant this as a comment for the PR to address. Are we fine commenting (or deleting) that line?

File: spec/system/admin/instrument_schedule_positions_spec.rb (hunk 3 above)
Comment (LeticiaErrandonea) [reply spec/system/admin/instrument_schedule_positions_spec.rb:28]: https://github.com/tablexi/nucore-open/blob/master/spec/support/matchers/display_order_matchers.rb

File: spec/system/admin/all_transactions_search_spec.rb (hunk 4 above)
Comment (LeticiaErrandonea) [reply spec/system/admin/all_transactions_search_spec.rb:34]: Oh, this shouldn't be here 😅

File: spec/system/admin/adding_to_a_cross_core_order_spec.rb [hunk 5]
```diff
@@ -215,18 +215,21 @@
         expect(page).to have_button("Add to Cross-Core Order")
       end
 
+      it "account for another facility cannot be selected" do
```
Comment (giladshanan) [inline spec/system/admin/adding_to_a_cross_core_order_spec.rb:218]: I'd add a comment here about why this got pulled out by itself

File: spec/system/admin/instrument_schedule_positions_spec.rb [hunk 6]
```diff
@@ -45,8 +47,12 @@
       find("[title='Move Up']").click
       click_button "Update Ordering"
 
-      wait_for_ajax
+      # Sometimes the first click doesn't work, so try again
+      expected_flash_message = "The Instruments have been reordered"
+      click_button "Move" unless page.has_content?(expected_flash_message)
```
Comment (giladshanan) [inline spec/system/admin/instrument_schedule_positions_spec.rb:52]: Should this be "Update Ordering"?

**Review by giladshanan — COMMENTED** [review]
Looks good - but I think there is one typo to address

**Pull request merged**